content/content/channels.md

```diff
@@ -0,0 +1,81 @@
+---
+title: Channels
+---
+
+# Channels
+
+Nim provides [channels](https://nim-lang.org/docs/channels_builtin.html), used to communicate between threads.
+
+Here is an example with plain threads.
+
+``` nimrod
+import std/os # for sleep
+
+var
+  # create a channel to send/recv strings
+  commChan: Channel[string]
+  sender: Thread[void]
+  recver: Thread[void]
+
+proc sendMsg() =
+  sleep(500)
+  # send a message in the channel
+  commChan.send("Hi")
+
+proc recvMsg() =
+  # block on the channel, waiting for output
+  let msg: string = commChan.recv()
+  echo "Received message: " & msg
+
+# very important: channels need to be opened, before they can be used
+commChan.open()
+createThread(sender, sendMsg)
+createThread(recver, recvMsg)
+joinThreads(sender, recver)
+```
+
+Usually, channels are created as global variables. See the [documentation](https://nim-lang.org/docs/channels_builtin.html#example-passing-channels-safely), if you wish to manually allocate shared memory for channels.
+
+The same example, using the higher level `spawn`:
+
+``` nimrod
+import threadpool, std/os
+
+var commChan: Channel[string]
+
+proc sendMsg() =
+  sleep(500)
+  commChan.send("Hi there!")
+
+proc recvMsg() =
+  let msg = commChan.recv()
+  echo "Received msg: " & msg
+
+commChan.open()
+spawn recvMsg()
+spawn sendMsg()
+sync()
+```
+
+Note that all messages are deep copied when sent.
+
```

Comment:
It'd make sense to have line 79 here.

Reply:
79 depends on 70-77. If you mean moving 70-79 to before 62, I can do that.

Reply:
Oh, I see. Looks fine the way it is!